test/rspamd_cxx_unit_cryptobox.hxx

```diff
@@ -0,0 +1,182 @@
+/*
+* Copyright 2024 Vsevolod Stakhov
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+/* Detached unit tests for the cryptobox */
+
+#ifndef RSPAMD_RSPAMD_CXX_UNIT_CRYPTOBOX_HXX
+#define RSPAMD_RSPAMD_CXX_UNIT_CRYPTOBOX_HXX
+#include "libcryptobox/cryptobox.h"
+
+TEST_SUITE("rspamd_cryptobox")
+{
+	/*
+	TEST_CASE("rspamd_cryptobox_keypair")
+	{
+		enum rspamd_cryptobox_mode mode = RSPAMD_CRYPTOBOX_MODE_NIST;
+		rspamd_sk_t sk;
+		rspamd_pk_t pk;
+
+
+		ottery_rand_bytes(sk, sizeof(sk));
+		ottery_rand_bytes(pk, sizeof(pk));
+
+		rspamd_cryptobox_keypair(pk, sk, mode);
+	}
+
+	TEST_CASE("rspamd_cryptobox_keypair_sig")
+	{
+		enum rspamd_cryptobox_mode mode = RSPAMD_CRYPTOBOX_MODE_NIST;
+		rspamd_sig_sk_t sk;
+		rspamd_sig_pk_t pk;
+
+
+		ottery_rand_bytes(sk, sizeof(sk));
+		ottery_rand_bytes(pk, sizeof(pk));
+
+		rspamd_cryptobox_keypair_sig(pk, sk, mode);
+	}
+*/
+	TEST_CASE("rspamd_cryptobox_hash")
+	{
+		rspamd_cryptobox_hash_state_t p;
+		const unsigned char *key = reinterpret_cast<const unsigned char *>("keys");
+		gsize keylen = sizeof(key);
+
+		ottery_rand_bytes(&p, rspamd_cryptobox_HASHSTATEBYTES);
+
+		rspamd_cryptobox_hash_init(&p, key, keylen);
+
+		const unsigned char* data = reinterpret_cast<const unsigned char *>("data");
+		gsize len = sizeof(data);
+
+		rspamd_cryptobox_hash_update(&p, data, len);
+
+		unsigned char out1[rspamd_cryptobox_HASHSTATEBYTES];
+		memset(out1, 0, rspamd_cryptobox_HASHSTATEBYTES);
+
+		rspamd_cryptobox_hash_final(&p, out1);
+		CHECK(*out1 == 'd');
+
+		unsigned char out2[rspamd_cryptobox_HASHSTATEBYTES];
+		memset(out2, 0, rspamd_cryptobox_HASHSTATEBYTES);
+
+		rspamd_cryptobox_hash(out2, data, len, key, keylen);
+		CHECK(*out1 == *out2);
+	}
+
+	TEST_CASE("rspamd_cryptobox_fast_hash")
+	{
+		rspamd_cryptobox_fast_hash_state_s *st = rspamd_cryptobox_fast_hash_new();
+
+		uint64_t seed = 10;
+
+		rspamd_cryptobox_fast_hash_init(st, seed);
+
+		const unsigned char* data = reinterpret_cast<const unsigned char *>("data");
+		gsize len = sizeof(data);
+
+		rspamd_cryptobox_fast_hash_update(st, data, len);
```

Comment:
We should do IUF in pieces to test something meaningful. We also need to check that our hash is matching our expectations.